linalg/mumps.cpp

```diff
@@ -0,0 +1,414 @@
+// Copyright (c) 2010-2020, Lawrence Livermore National Security, LLC. Produced
+// at the Lawrence Livermore National Laboratory. All Rights reserved. See files
+// LICENSE and NOTICE for details. LLNL-CODE-806117.
+//
+// This file is part of the MFEM library. For more information and source code
+// availability visit https://mfem.org.
+//
+// MFEM is free software; you can redistribute it and/or modify it under the
+// terms of the BSD-3 license. We welcome feedback and contributions, see file
+// CONTRIBUTING.md for details.
+
+#include "../config/config.hpp"
+
+#ifdef MFEM_USE_MUMPS
+#ifdef MFEM_USE_MPI
+
+#include "mumps.hpp"
+
+namespace mfem
+{
+
+void MUMPSSolver::SetOperator(const Operator &op)
+{
+   auto APtr = dynamic_cast<const HypreParMatrix *>(&op);
+
+   MFEM_VERIFY(APtr, "Not compatible matrix type");
+
+   height = op.Height();
+   width = op.Width();
+
+   comm = APtr->GetComm();
+   MPI_Comm_size(comm, &numProcs);
+   MPI_Comm_rank(comm, &myid);
+
+   auto parcsr_op = (hypre_ParCSRMatrix *) const_cast<HypreParMatrix &>(*APtr);
+
+   hypre_CSRMatrix *csr_op = hypre_MergeDiagAndOffd(parcsr_op);
+#if MFEM_HYPRE_VERSION >= 21600
+   hypre_CSRMatrixBigJtoJ(csr_op);
+#endif
+
+   int *Iptr = csr_op->i;
+   int *Jptr = csr_op->j;
```

Comment:
These two lines are failing for me since I have hypre built with `--bigint` and therefore `HYPRE_Int` == `long long int*`. If MUMPS supports 64-bit matrix indices, we should also support that, otherwise a compile time error should be printed that there is a conflicting configuration of hypre.

Reply:
good point! I'll fix it

Reply:
MUMPS by default supports 64-bit ints only for the `nnz_loc` counter.  In case we want full support for 64-bit ints, MUMPS should be compiled with C preprocessing flag `-DINTSIZE64` and Fortran compiler option `-i8` (or equivalent). Additionally, all the required libraries have to be compiled with 64-bit support (MPI, BLACS,ScaLAPACK, LAPACK , BLAS etc). I am not sure that this is crucial, at least for now, so I decided to add a check for Hypre BIGINT (same check we do for SuperLUdist), which will print a compile time error. Sounds good?